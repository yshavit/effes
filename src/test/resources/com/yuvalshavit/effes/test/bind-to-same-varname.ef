def foo (bool: BoolWrapper | VoidWrapper) -> True | False | Void:
  return case bool of
    BoolWrapper(v): v
    VoidWrapper(v): v

def main -> Void:
  print (foo BoolWrapper(True))
  print foo VoidWrapper(Void)
  return Void