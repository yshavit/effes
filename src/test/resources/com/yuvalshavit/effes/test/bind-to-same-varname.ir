def foo (bool: BoolWrapper | VoidWrapper) -> False | True | Void:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: bool
          pos: 0
      of BoolWrapper:
        VarExpression
          var: v
          pos: 0
      of VoidWrapper:
        VarExpression
          var: v
          pos: 0
