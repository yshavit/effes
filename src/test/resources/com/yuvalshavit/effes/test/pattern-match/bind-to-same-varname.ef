foo (bool: BoolWrapper | VoidWrapper) -> True | False | Void:
  return case bool of
    BoolWrapper(v): v
    VoidWrapper(v): v

main -> Void:
  debugPrint (foo BoolWrapper(True))
  debugPrint foo VoidWrapper(Void)
  return Void
