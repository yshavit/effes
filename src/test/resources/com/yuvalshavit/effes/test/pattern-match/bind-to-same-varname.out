Your argument was: True
Your argument was: Void
>> Void
